src/em/prov/easyconnect/ec_ctrl_configurator.cpp

```diff
@@ -296,29 +298,211 @@ bool ec_ctrl_configurator_t::handle_proxied_dpp_configuration_request(uint8_t *e
+    size_t bsta_config_object_len = cjson_utils::get_cjson_blob_size(bsta_config_object);
+
+    // Create DPP Configuration frame.
+    auto [frame, frame_len] = ec_util::alloc_gas_frame(dpp_gas_action_type_t::dpp_gas_initial_resp, session_dialog_token);
+    ASSERT_NOT_NULL_FREE(frame, false, wrapped_attrs, "%s:%d: Could not allocate DPP Configuration Response frame!\n", __func__, __LINE__);
+    ec_gas_initial_response_frame_t *response_frame = reinterpret_cast<ec_gas_initial_response_frame_t *>(frame);
+
+    uint8_t *attribs = nullptr;
+    size_t attribs_len = 0;
+    attribs = ec_util::add_wrapped_data_attr(reinterpret_cast<uint8_t *>(response_frame), sizeof(ec_gas_initial_response_frame_t), attribs, &attribs_len, true, e_ctx->ke, [&]() {
+        size_t wrapped_len = 0;
+        uint8_t *wrapped_attribs = ec_util::add_attrib(nullptr, &wrapped_len, ec_attrib_id_enrollee_nonce, e_nonce_len, e_ctx->e_nonce);
+        wrapped_attribs = ec_util::add_attrib(wrapped_attribs, &wrapped_len, ec_attrib_id_dpp_config_obj, static_cast<uint16_t>(ieee1905_config_obj_len), reinterpret_cast<uint8_t*>(ieee1905_config_obj));
+        wrapped_attribs = ec_util::add_attrib(wrapped_attribs, &wrapped_len, ec_attrib_id_dpp_config_obj, static_cast<uint16_t>(bsta_config_object_len), reinterpret_cast<uint8_t*>(bsta_config_object));
+        return std::make_pair(wrapped_attribs, wrapped_len);
+    });
+
+    response_frame = reinterpret_cast<ec_gas_initial_response_frame_t*>(ec_util::copy_attrs_to_frame(reinterpret_cast<uint8_t*>(response_frame), sizeof(ec_gas_initial_response_frame_t), attribs, attribs_len));
+    if (response_frame == nullptr) {
+        printf("%s:%d: Failed to copy attributes to DPP Configuration frame!\n", __func__, __LINE__);
+        free(attribs);
+        free(wrapped_attrs);
+        free(response_frame);
+        free(ieee1905_config_obj);
+        free(bsta_config_object);
+    }
+    response_frame->resp_len = static_cast<uint16_t>(attribs_len);
+
+    // If a Multi-AP Controller receives a Proxied Encap DPP message from an Enrollee Multi-AP Agent carrying a DPP
+    // Configuration Request frame, it shall generate a DPP Configuration Response frame and include one DPP Configuration
+    // Object for the 1905-layer and one DPP Configuration Object for the backhaul STA of the Enrollee, encapsulate them into
+    // a 1905 Encap DPP TLV, set the DPP Frame Indicator bit to one, set the Enrollee MAC Address Present bit to one, set the
+    // Frame Type field to 255 and include the Enrollee MAC Address into the Destination STA MAC Address field. 
+    auto [encap_response_frame, encap_response_frame_len] = ec_util::create_encap_dpp_tlv(true, src_mac, ec_frame_type_easymesh, reinterpret_cast<uint8_t*>(response_frame), frame_len + attribs_len);
+    ASSERT_NOT_NULL_FREE(encap_response_frame, false, wrapped_attrs, "%s:%d: Failed to alloc DPP Configuration frame!\n", __func__, __LINE__);
```

Comment:
Need to free response frame before assert (and it's fine at this point since it's been encapsulated)
```suggestion
    auto [encap_response_frame, encap_response_frame_len] = ec_util::create_encap_dpp_tlv(true, src_mac, ec_frame_type_easymesh, reinterpret_cast<uint8_t*>(response_frame), frame_len + attribs_len);
    free(response_frame);
    ASSERT_NOT_NULL_FREE(encap_response_frame, false, wrapped_attrs, "%s:%d: Failed to alloc DPP Configuration frame!\n", __func__, __LINE__);
```